internal/dataplane/translator/translator.go

```diff
@@ -414,10 +415,24 @@ func (t *Translator) getUpstreams(serviceMap map[string]kongstate.Service) ([]ko
 					}
 				}
 
-				// add the new targets to the existing pool of targets for the Upstream.
-				targets = append(targets, newTargets...)
+				for _, t := range newTargets {
+					// See https://github.com/Kong/kubernetes-ingress-controller/issues/5761:
+					// Duplicate targets will appear in configurations that use Services with the same selector, which are used
+					// by some rollout systems. We need to deduplicate them while honoring the total weight.
+					//
+					// Because kongstate.Target is a nested kong.Target and the target IP is also a field named Target, the
+					// key names are a bit silly.
+					if existing, ok := targetMap[*t.Target.Target]; ok {
+						sum := *existing.Weight + *t.Weight
+						existing.Target.Weight = &sum
```

Comment:
Apart from naming there's `existing.Weight` and `existing.Target.Weight` which is not immediately clear what's what here.

Do you mind clueing the readers on what's happening here?